### ISSUE TITLE
End to End Testing Recomendation

### ISSUE THREAD

**Issue by MikeWalrath** [hpe-design/aries#54]
1. Aries core (similar to Grommet)
- Testing per component 
- Accessibility testing (package exploration) 
- Storybook
- Chromatic 

2. Aries site
- End to end (layout, content, routing, navigation and deep linking)
- Cypress
- Accessibility 
- SEO (header/alt tags)

**Comment by taysea** [comment]
Going to leave my thoughts on testing below and update if I look into any of the other bullets:

aries-core:
- For accessibility testing, I propose we use `jest-axe`. This package “wraps axe-core for use in jest tests”. To see all the accessibility rules tested, see here: https://github.com/dequelabs/axe-core/blob/HEAD/doc/rule-descriptions.md. In our tests, we would create one for accessibility and use jest-axe to implement something along the lines of `expect().toHaveNoViolations`. If any of the tested rules fails, we will be flagged and know that a prop or other attribute needs to be added.

aries-site:
- Cypress can be used for integration testing as well as accessibility testing on the site. Cypress itself does not have a tab event (there are issues open since 2016 asking for this). However, there is a separate plugin someone has built (https://github.com/Bkucera/cypress-plugin-tab) but it may not be 100% reliable.
